core/src/SYCL/Kokkos_SYCL_Space.cpp

```diff
@@ -75,17 +89,20 @@ SYCLHostUSMSpace::SYCLHostUSMSpace()
 SYCLHostUSMSpace::SYCLHostUSMSpace(sycl::queue queue)
     : m_queue(std::move(queue)) {}
 
-void* allocate_sycl(
-    const char* arg_label, const size_t arg_alloc_size,
-    const size_t arg_logical_size, const Kokkos::Tools::SpaceHandle arg_handle,
-    const RawMemoryAllocationFailure::AllocationMechanism failure_tag,
-    const sycl::usm::alloc allocation_kind, const sycl::queue& queue) {
+void* allocate_sycl(const char* arg_label, const size_t arg_alloc_size,
+                    const size_t arg_logical_size,
+                    const Kokkos::Tools::SpaceHandle arg_handle,
+                    const sycl::usm::alloc allocation_kind,
+                    const sycl::queue& queue) {
   void* const hostPtr = sycl::malloc(arg_alloc_size, queue, allocation_kind);
 
-  if (hostPtr == nullptr)
-    throw RawMemoryAllocationFailure(
-        arg_alloc_size, 1, RawMemoryAllocationFailure::FailureMode::Unknown,
-        failure_tag);
+  if (hostPtr == nullptr) {
+    using FailureMode =
+        Kokkos::Experimental::RawMemoryAllocationFailure::FailureMode;
+    Kokkos::Impl::throw_bad_alloc(arg_alloc_size, std::align_val_t{1},
+                                  FailureMode::Unknown,
```

Comment:
I preserved the current behavior but this will need fixing.
Apparently `FailureMode::Unknown` was more intended as an invalid state if we look at the definition of `RawMemoryAllocationFailure::what()`.